### ISSUE TITLE
How to execute a single non Auto “Mow task” without going into Auto mode?

### ISSUE THREAD

**Issue by BionicThomas** [dinmammas/homebridge-robonect#7]
Thanks for the plug-in!

From time to time I would like to send the mower to mow for X hours and then return to base in „Home“ mode to wait for the next task.
So not in Auto Mode.
Is there a way to achieve this already today or would it be possible to add such a switch?

I‘m using the „Steuerung“, „Neuer Mähauftrag“ Option on the robonect Web GUI.
Would be great to allow to configure the options provided on this form in the JSON and then have a switch to run this command.

![912385AD-EC27-4432-9B65-E408CBB4D882](https://user-images.githubusercontent.com/50406462/124378569-54cde080-dcb2-11eb-8979-f0221488fb12.jpeg)

**Comment by dinmammas** [comment]
Hello! 
Thanks for the suggestion!

Fastest way to achieve this, is to take a look at this plugin: https://www.npmjs.com/package/homebridge-http-switch
Specifically a stateless switch.
You could configure multiple stateless switches for multiple jobs.

Best,
David

**Comment by dinmammas** [comment]
@BionicThomas 

Don't know if you've already tried it, but here's a URL that should work with the aforementioned plugin. (change the user/pass/server/port ofc)
http://user:pass@server:port/json?cmd=mode&mode=job&end=13:00&after=auto&remotestart=1

The URL should start the mower immediately from remote start 1, and return to auto mode after 13:00.

I've not used the "job" feature myself, but the API docs says:

> ....job <Mäher führt einen Mähauftrag aus>
> ........duration <Dauer des Auftrages. Wenn der Mäher lädt, läuft die 'Uhr' dennoch!>
> ........start <Startuhrzeit 'hh:mm', kann entfallen, dann gilt 'sofort'
> ........end <Endzeit 'hh:mm'>
> ........after <Modus, der nach diesem Mähauftrag aktiviert werden soll. '0' bedeutet, den Modus nicht ändern>
> ........remotestart <Fernstartpunkt; 0:Ladestation, 1:Startpunkt 1, 2: Startpunkt 2; evtl. <Danke an demichve> 0:Standard(?), 1:Normal(?), 2: Pkt 1, 3: Pkt 2>

GLHF ;)

**Comment by BionicThomas** [comment]
Many thanks!
Started to look at the RoboNect forum to get started but was not that far yet.
I will try to supply a duration only without and end time. This button might be used at any time of the day hence the fixed end time would not be ideal.
Again many thanks!

**Comment by BionicThomas** [comment]
Hi! Just wanted to complete the loop.
I was able to achieve what I wanted via your suggested "http-switch" plugin.

{
            "accessory": "HTTP-SWITCH",
            "name": "Mähe 1 Stunde",
            "switchType": "stateless",
            "timeout": 1000,
            "onUrl": "http://<USER>:<PASS>@<IP>/json?cmd=mode&mode=job&begin=0&duration=60&after=0&corridor=255&remotestart=0"
        }

My wife is now happy to use the mower in manual mode via this command via Siri.
Added two more versions for 2 and 3 hours.
Just keep in mind that the duration counter starts instantly and in case the mower is charging, the remaining charging time is reducing the actual mowing time before it starts to mow.

Many thanks!
